package.json

```diff
@@ -48,6 +48,7 @@
     "@types/node": "^17.0.8",
     "@types/react": "^17.0.38",
     "@types/react-dom": "^17.0.11",
-    "typescript": "^4.3.5"
+    "typescript": "^4.3.5",
+    "use-debounce": "7.0.1"
```

Comment:
A nice implementation of the debounce concept that allows the search to only query when the user finishes typing.  The O.G. version I had was searching on every keystroke, which is a horrible policy lol